src/object_properties.cpp

```diff
@@ -171,4 +173,5 @@ void ObjectProperties::deSerialize(std::istream &is)
 	try {
```

Comment:
I suspect this try/catch block was added for backwards compatibility, to avoid changing protocol version. Should I do the same? It looks awkward.

Reply:
Yes, you're right. Attempting to deserialise properties within a `try-catch` allows clients to silently skip deserialising a property if an older server didn't send it. So please move L176 into the `try` block.

Reply:
There's an issue here:
`readU8` doesn't throw exceptions but returns zero when there was no data to read, so if you were to join an older server shading would be disabled for all entities.

Your options are to either send the value inverted or to add this wonderful construct:
```cpp
    u8 tmp = readU8(is);
    if (is.is_eof())
        throw SerializationError("");
    shaded = tmp;
```

Reply:
I moved it out of try/catch block and reworded a bit

```c
	u8 maybe_shaded = readU8(is);
	if (!is.eof())
		shaded = maybe_shaded;
```

Reply:
If `deSerializeString` throws an exception the rest shouldn't run, so it does need to be in the try-catch.

Reply:
Fixed.